syntax = "proto3";

package ragllm_service;

option go_package = "github.com/milkyhoop/flow-executor/internal/proto/ragllm_service";

import "google/protobuf/empty.proto";

// Layanan utama
service RagLlmService {
  rpc DoSomething (Ragllm_serviceRequest) returns (Ragllm_serviceResponse);
  rpc HealthCheck (google.protobuf.Empty) returns (google.protobuf.Empty);
  rpc GenerateEmbedding (EmbeddingRequest) returns (EmbeddingResponse);
  rpc SearchSimilarDocuments (SearchRequest) returns (SearchResponse);
  rpc GenerateAnswer (GenerateAnswerRequest) returns (GenerateAnswerResponse); // ✅ Baru
}

// Request & Response
message Ragllm_serviceRequest {
  string user_id = 1;
  string input = 2;
}

message Ragllm_serviceResponse {
  string status = 1;
  string result = 2;
}

message EmbeddingRequest {
  string text = 1;
}

message EmbeddingResponse {
  repeated float embedding = 1;
}

message SearchRequest {
  string tenant_id = 1;
  repeated float embedding = 2;
  int32 top_k = 3;
}

message SearchResponse {
  repeated SearchResult documents = 1;
}

message SearchResult {
  string id = 1;
  string title = 2;
  float score = 3;
}

// ✅ Tambahan untuk GenerateAnswer
message GenerateAnswerRequest {
  string question = 1;
  string tenant_id = 2;
  string mode = 3; 
}

message GenerateAnswerResponse {
  string answer = 1;
  string mode = 2;
}
